airflow/contrib/operators/gcp_bigtable_operator.py

```diff
@@ -0,0 +1,424 @@
+    def __init__(self,
+                 project_id,
+                 instance_id,
+                 main_cluster_id,
+                 main_cluster_zone,
+                 replica_cluster_id=None,
+                 replica_cluster_zone=None,
+                 instance_display_name=None,
+                 instance_type=None,
+                 instance_labels=None,
+                 cluster_nodes=None,
+                 cluster_storage_type=None,
+                 timeout=None,
+                 *args, **kwargs):
+        self.project_id = project_id
+        self.instance_id = instance_id
+        self.main_cluster_id = main_cluster_id
+        self.main_cluster_zone = main_cluster_zone
+        self.replica_cluster_id = replica_cluster_id
+        self.replica_cluster_zone = replica_cluster_zone
+        self.instance_display_name = instance_display_name
+        self.instance_type = instance_type
+        self.instance_labels = instance_labels
+        self.cluster_nodes = cluster_nodes
+        self.cluster_storage_type = cluster_storage_type
+        self.timeout = timeout
+        self._validate_inputs()
+        self.hook = BigtableHook()
+        super(BigtableInstanceCreateOperator, self).__init__(*args, **kwargs)
+
+    def execute(self, context):
+        instance = self.hook.get_instance(self.project_id, self.instance_id)
+        if instance:
+            # Based on Instance.__eq__ instance with the same ID and client is considered as equal.
+            self.log.info(
+                "The instance '%s' already exists in this project. Consider it as created",
+                self.instance_id
+            )
+            return
+        try:
+            self.hook.create_instance(
+                self.project_id,
+                self.instance_id,
+                self.main_cluster_id,
+                self.main_cluster_zone,
+                self.replica_cluster_id,
+                self.replica_cluster_zone,
+                self.instance_display_name,
+                self.instance_type,
+                self.instance_labels,
+                self.cluster_nodes,
+                self.cluster_storage_type,
+                self.timeout,
+            )
+        except google.api_core.exceptions.GoogleAPICallError as e:
+            self.log.error('An error occurred. Exiting.')
+            raise e
+
+
+class BigtableInstanceDeleteOperator(BaseOperator, BigtableValidationMixin):
+    """
+    Deletes the Cloud Bigtable instance, including its clusters and all related tables.
+
+    For more details about deleting instance have a look at the reference:
+    https://googleapis.github.io/google-cloud-python/latest/bigtable/instance.html#google.cloud.bigtable.instance.Instance.delete
+
+    :type project_id: str
+    :param project_id: The ID of the GCP project.
+    :type instance_id: str
+    :param instance_id: The ID of the Cloud Bigtable instance to delete.
+    """
+    REQUIRED_ATTRIBUTES = ('project_id', 'instance_id')
```

Comment:
Remove `project_id`  from `REQUIRED_ATTRIBUTES` and have an option to use `project_id` from connection

Reply:
Can we do it in separate commit making it consistent for all other operators? I think it make sense to do it - but then doing it in one commit when we fix it for other operators might make more sense I think. Unless you think it's better to do it now ? we will do it within few days otherwise.

Reply:
This is a new operator, hence I would still feel as it is just a matter of removing an attribute. However, I don't have a strong feeling about it - so it can wait for the next PR